### ISSUE TITLE
Tripal 3 legacy API has diverged from Tripal 2 -Bring it back up to date

### ISSUE THREAD

**Issue by spficklin** [tripal/tripal#72]
Copied from Drupal isssue queue: https://www.drupal.org/node/2713125

Various bug fixes have been made to Tripal 2 since Tripal 3 development began. These bug fixes should be merged into the Tripal 3 API where applicable. Due to the massive changes this will have to be done on a per commit bases...

According to this comparison this will be a doozy of a job :-(
https://github.com/tripal/tripal/compare/7.x-3.x...7.x-2.x

Let's try to keep track of it as we go in this issue.

@laceysanderson  reported some progress on this.  See Drupal issue queue for further details.

**Comment by laceysanderson** [comment]
In order to go through it commit by commit, we would have to check the following:
- [x] cc79e48963514069c545d9c36625e58b5ba3f9f9
- [x] 43077a9f78b5fb68017007de9e770b5345b8fa50
- [x] a5ca25cd66c73402a341651d0103deecb97bccac
- [x] f6a519777358f23f73cae0c35119aa3f6a425ba0
- [x] 4c8dc943518b375f4f9812496d094b5a04979fb8
- [x] 9a34e7512e98d290e749ac8c780d35a4a4c980a9
- [x] eb19a57b02a8842ffbc043489492f2c7145542b2
- [x] 2ad7497a2d04150a5f9f8abf424876d27a405063
- [x] 9f42bd0f5dc5094559fa2f2ee63c06214e85e942
- [x] d8a67ee2a3f1ea428e4d6f0dca6fab5ba1866753
- [x] 5f840f3092bb60f268f39940aab63f7489675478
- [x] dd51ce667cd2a36ea875eaeac717150ca3a21b50
- [x] 4baefc955881c964897ef0dccdf102149f074817
- [x] 44e414bae62661c6e4ccef43ad794a8042bc1d0b
- [x] 0781d162e5aa5aa305fc91e992feedd2a0b6985e
- [x] ae32d1493c47a6d85cb4640715afd8f47c327a1d
- [x] 7573ad21c59ff586b50d093a77a0110730c47e95
- [x] eb8998689038d03e0da995e7f9f70b67d9cfebd9
- [x] a89f289b111b985725ce5f9851a5847391852531
- [x] 78846a20303aff46db3013e250d3616e86e4e0de
- [x] 3ca8ef2d99dcb2996e031e0310a62734b387b7d6
- [x] 8db82989d63e8388344b91de191fe040bcc78df2
- [x] 9b2110f76939b7825e61a46a433686953c47b8ba
- [x] 38b36aa2b6a0cd7e80baf049235a89dd2aabc2ba
- [x] f7cd85c2f83a0883be8a680341c3e085330fa2e3
- [x] a6efcd56005b710df1097903178c26dc69205bde
- [x] 00f95e05b6ebaa110a9cabd37d627febf3ea0d31
- [x] fc7fdd576fa27ed9813ea20810149d7278d5b3d9
- [x] 18cd56a8a3214cbabf1c8edbe1c02cecab032d0a
- [x] 5b23ea4a20ef71168d64128fe0a6bac8a3561e1d
- [x] d13dac462761a06b6e8e4239f5863d1966cfdf86
- [x] 6963a7659d3460b6233a19dc53842ebbc466ebdf
- [x] 3497ab31d7ee749da872f6f164122f321c75be64
- [x] 78199e108d6bf2d2e77e82e49511adf049c09877
- [x] 49f10141e27e18ea634e586a01eb43d040455950
- [x] 32ada67f2337c25f5b4d73296a146a64e9072ba4
- [x] 6e666058bbcf80efa640f66bda982c8d317960f4
- [x] f7c23ae92c8018b7d4011e7ba163e3c758b5cf66
- [x] 66994a6d40800630cfcf037f6b212675ec9c25d3
- [x] b302998a1f5c6f60ace5943db836f4735fccbd74
- [x] 47a7c1a955cf4fc4ed690ba6f4b79a5bb3afafb7
- [x] a6c0e4d912022b9cc954221d59941f0c1121e9b7
- [x] c099ef7f50bf67e889cd90b1cd59baf13f0554ab
- [x] 6ae567ca6f4fd1a4f9c2dcf9f2582c424382e0ab
- [x] 5a52f079ef85d854b6cad7e6bdec44009026a753
- [x] 15ad1ea90a9bce30c7e1eb72526bd8eaaf7c7ac7
- [x] 468ed6fb931013b5def583a06cd0393e2d885da1
- [x] dcd6ca91140608ddd26e5dded1bf1ec6f0a18f92
- [x] e933d1946cbd46eab0c1290c8c990254de79bb38
- [x] 77c960eabbe84ac481d3de3cc49d04310182b748
- [x] 5c169efeb9b363b7489d56ba55a452bcc0cb113f
- [x] 4eb43e4db3869fba382b1ae4ac0797aa69ba33ff
- [x] 6221816fb002b3fb7c3e644d0043ec471543a956
- [x] 3e8f4ef595cc821a16d4e0f44423490d5192555b
- [x] d1d2cdc4a860046c6440657c9341b8e4b53abaa5
- [x] 2b229c02bc7644ea91ae025bf2faad5681950875
- [x] 573dfa42369b1735792fd9e9549342361e465a85
- [x] 3cafe940dfa06fd1b5407414645447d2d1257aa7
- [x] 48bb8b43ca49567985a52ca350ffe9bfa7023748
- [x] 0b743046a0785b951b54e61edc671e832baf0cf0
- [x] e2be20af4ce991d12dcbfb95664c9c2c19588560
- [x] 9bde2de126ff8c9eb8364dff801da1cbae8549b6
- [x] cf5096dd0f1415a87de5f1f7fdc8da5922747d8a
- [x] 10780453ac4fb13e43821839ce47a2377e7a2a30
- [x] 2073a4bcb504e643e762cc8506dc4eeffd8f71e9
- [x] 80e5a11ff89897328863f7e9bc6d93530941aa02
- [x] e1987960d8c62a22d816cbca311e9246987df116
- [x] ef913958bd830405ed4c53f1ef9f89be9e97d9b6
- [x] f860f87c06808a96dc054a256d89a5f62a6001ab
- [x] 7765e11a265f2aa0e89cdaff9b6b73598f6c5d12
- [x] e89b4bdcb43f03790cbdd8603668013c038a704e
- [x] 143635ccf0bafcec91083d4d5d1058249f36f411
- [x] 95a6a16e79cc64822ad07003caa657059aa76004
- [x] b26aa147510dcf1a139d7608d2db5215e14f1d8e
- [x] 3970318ebeb37b91fb9054bc4ecb0084c4401b81
- [x] 2ecc757cc822a89641877d6341f4964140113248
- [x] 1f277a436e6ffcb3952f1b9be788560fbd8c5390
- [x] 56fa9a34465a3cd0eb991f2f57c4cc09d25e78b5
- [x] 8ca88432d1d737ae5d84a69e440504aa0feeeaf8
- [x] f438dec3b74ee9a6132213f1a69815e99207fa59
- [x] 847facc2365ef0231b6664997368e0f71d71352c
- [x] 4cfc841a5285b45a2712bb024d5949cda433a91c
- [x] 74b9f5b95f7796b27e99ee9f6046ff7a5ceaa83e
- [x] 4d108d7de4a1f023a3be28a93986a4f4f6603e9a
- [x] 5cd756da7f0212e5835e89fa25f5d15e2af50a48
- [x] 129ef89289b6b52642b4575e4a057d1f57d2a15b
- [x] 7f1c225098b92be2ff7e215c9df9ccfe754adc72
- [x] 5d9d10d9c488d13a4c53a508b9df1b9d135899a7
- [x] a697fcf0b33b50834a17fd35cf2ca8b53db8df50
- [x] 9e1c7d65cdef1b324bdef050a87556e3e39bb3ae
- [x] dbee73dab814e3fa0563ddc7d34ef4a3074a7069
- [x] d3cb449f5c6ab2b625b3774208efa476dbb04553
- [x] 8fa6fdac0665bbad5f719d7fff571be9ae658d34
- [x] eb18d3a9f03b40c37e8c425b2e45cad67da88045
- [x] 19bb1ba2ad687a708ba37d1396aa817b9a508f16
- [x] e8a568d39b48797c8ecd4649906ecc8bc5949bfe
- [x] bea68e15196111ce23190320f6a2a16fa740ffb7
- [x] e6e9a5e2e5fdca50bf8875eca184da7986a089e4
- [x] 9ed3b6ddc6c383132d06ac449743d493055278e8
- [x] 8832745297336d7143328dee48c55e2faa1a6f17
- [x] 16c120e2da82184845fe281a673a8ac692b4f721
- [x] 6d6e81a4f05255ee7500171d3094956e01997a7a
- [x] 23e075f216daf9646637a633e272e24381778722
- [x] d664e90ab0fac73b9653492b5448e55dae15b743
- [x] 4f0ce3b284a360e990d22eb360b3cbf2ce5854c3
- [x] fb0379a4b29c4f4acad44b80260caa71c2c3332a
- [x] c82c25f80e5aa18c6b550435b6e47d7574b76bf5
- [x] 19710e02b6a864f1e89ab996cfbb164d5c9af2de
- [x] 04e3b6b8abc32985e531deaa47d69fa4947a254c
- [x] c624e39393520bad9f53f203d749b8c3c3326b1b
- [x] e4a085d523e30651c4a053040cf95c4d8da44e1a
- [x] ffe19370a320c9026995dacda3d698579c8b515d
- [x] e8e4a8dca74109b16f300e1db5b8e585bbafdca6
- [x] d5c65cec3aa56c6ae208a896fc9da8522e60a579
- [x] 10ad767e714b53b55efd0d98aa4aabb9fe785239
- [x] e64575d83a8db0fc2eb4fb1b967ad0002bf099fd
- [x] 52316c316619043dd2f56597e64e6eb0701271c0
- [x] 14cb0d52b64e5f2795cefce28d23618caa0d93cc
- [x] ae14e34547df9fc17437389075dd4d1c62290315
- [x] eca67392b7f19057113eefb5e518a0394926652b
- [x] 77062152c2f528e0d7f8990a54be71940d0b892c
- [x] a06c4e411d28c9440e549c063ddaacaef28da05d
- [x] d9ab118b47a35c8cca0908fc6f49bd16b020ca17
- [x] f946a106ffc78f47a1bf44c00a9425d9e1d22685
- [x] 46bf77a31f27f474c46682124a5e9d41c089af58
- [x] 9cc0b65150d56e10f824645ff77d99a2b977d645
- [x] 49dfcad26e8906b915c43cf872310e1cb08549a7
- [x] af0fd6afcec0f8bdc0d511b49611d9902d24dd80
- [x] 429efdd20faa748f7ca63c476793d4847919b542
- [x] 4ccf75197bc5fe6435aeed12bc2be626efe3b8fb
- [x] 69721083ca91728fdec2732b8fa4555003ff0922
- [x] 5e699c06965f652e12b41043d01f8c167f3c42d8
- [x] b2b261147ddae32076f77c0bda9b39d5ded22315
- [x] ecfec7030844244f20775809b7c17f563aa19c19
- [x] 8758efcdd54172db5b609a401621f02a469995d6
- [x] 13921c293463551daabc0797904514c30a2ed80b
- [x] 4cb057df9294033d9e642ffe61832becc745f6f9
- [x] 66a25d58a15057dbeb40b989c66bd9c8a26cfbed
- [x] 895f8b07fa840509ea98747b8425485e096da2c4
- [x] 0ff52bd8681981dc11abae029640c5868b23450f
- [x] c8153a16087ac2f2851b6e719dab2e01a6e14e49
- [x] cebc3dfb332a9d1a0d045a7eb57c7b17b63a6875
- [x] 78eb368b516f30b1c8f87685dee26141b071e8f4
- [x] 7d609ad9e9bc8e117301aa82f126c3da30ff6218
- [x] dfa8c59756b08a627a513893e8bdef6818eebb58
- [x] 97661c6ffc2ff284b6a08d23bb09167e8839efdd
- [x] 124106bc43a702d6ea425beeb9b87daaf4b8ae52
- [x] cee9402ff659ec245e651601554655da79bdc25b
- [x] 1bf62fee22483e2a1192d7a030c4323abcb2c963
- [x] bb0d7c55c91ab40ed151a2834dd6a3953b97400b
- [x] 84828e05e2c7d612a1929e5fa41b3748c3a8cd71
- [x] 878cb904e7ede62a39cd699071f528581460bfee
- [x] f297d0723f2c9557ce47723a930f0d923971aa70
- [x] 4fcd8a60549ec638c4edeb4999c94397a2d2da55
- [x] a656e9da6778c4ce6f90d85ac075522fd9363c80
- [x] 9fc326e10ac4b34fd3105c9abf86c7bb63cb7901
- [x] fd295877caf591bf1b58adda58e8348c8a2d0072
- [x] eb473ced496e04aafbf09717ef454e08a6ebc07d
- [x] 8b8ff48ba4156a5e1b75a997e46b5faac2f26145
- [x] e348ad399da4713d59f2d03c5ab4af7aafe80a8a
- [x] 36f766b02397b99e28c32d0dcaad796d37950e7b
- [x] 55f65cba1ccf7dce6bfd4731da388282ba5cf0dd
- [x] 64a33e6e53c60bbf25cbc60ea92d737d3355252b
- [x] a75d8b31d08f3e45f738194ab1d3ceff85d56f1c
- [x] 2b9f466e678cf8b9ff66e22848f9ed6e7405ff9b
- [x] 3e42625d8b3adb302b24fedefea7e1b148522b0a
- [x] e7531954d0b640d81631ea91a4564ceb4bcc9ea9
- [x] 736ce8956f0762eb133c5d3313e860b7161e6270
- [x] 4beb60a240764a5edbf7e8eae7bed6ecf9336da7
- [x] a496de3a561a1933277f70c638d8dc1c785457c4
- [x] 475ab415293d7d6d6aee228caefab1e13bc15dd2
- [x] 1d1e2360f1d2e016a0cd386eafb723460654045a
- [x] f0c3f925e1338c6f5d04d9078b23f8d01c7edf04
- [x] e5cbf15662e7c4000f9cb5d158c7ab8c3d1a62c9
- [x] 1dd0273d3b079c5ae3a5c72d3e7985a7a614c03b
- [x] 896d8a8454d2ad789115913c8a5269969eb4870f
- [x] fa3fdb49cce417fd91b1741335d647e596002827
- [x] b8cb8098697509e3a627a9ef11091eea0d395c34
- [x] 03370c5db88e3fb3a06476fc1fe05447b3f48400
- [x] 04fdc885fdc5a9941cc0cc5119c543cc63ecf1c4
- [x] a48adff100cb2a528c4abcc25b3e0aca5b7fdf32
- [x] 83574d5d147ba2a0841fc96b7c92b2f1e44fe7e4
- [x] 154e72b4fd9b80fc382594a85add90f8fc02585e
- [x] f5d965b4209eef126830539816c293c2256cd8d1
- [x] ad9e48bb9c3a3984e5bb2145d425823d0a784fba
- [x] 32623d26eccfbe6db3792f9bca568741622dd3bf
- [x] afe432ddb14e928d3062bb724179da7d1180d20c
- [x] c7d0a99c600bf5bde4e920b4511544d942929718
- [x] c8155e57202453949911fe97ddff89fe4f3ba86c
- [x] 8f79927f4b0ffb92f3fd851e036680731824deb0
- [x] c4d24cee5da8b8ce9013eab87b1e525665c96367
- [x] 16c1861d7a23d83222a99839e5f23cac386f735f
- [x] 4de88a0aa19fd73fea525670873895358d1052d1
- [x] beb208b33f6aa34feb1880ed7e7a8807022c6c51
- [x] 876b9cdae62e6f2332659fab19d6676fe0f35e79
- [x] 29eb480a47e91db205a95bf3c28712d4530d535c
- [x] 9a05f034b4f4e371d6f070374105221961c86d3a
- [x] a48bf3b869ec7144b4f75390260123edf28b4b07
- [x] 772ede09ca13423da47e9d920ac3c5347bf76035
- [x] 4297f83cc84316b35a0d300bbc6cc84e2ba36f4f
- [x] 24567377ce3a09a19056e0579817f55dd5fdd191
- [x] cc37006f033cc9bd7426ff300d7adf3be9a1216b
- [x] f5f7195ef503f5a8a4e6e538f5a84d22f025d327
- [x] cfe6848e011e5f524459ae5b0a0ba6a81d5fc94b
- [x] 9dee16a212e112d9db8b2656c578ee919c2a4fdf
- [x] 2c19537660eac88f8b16793254f7ec2896361632
- [x] 6d07c4f9ba2aab0b787b6d04e004771a1ceb313c
- [x] 74e4f6f717e720fa077fcdd8dda8745b04ac7739
- [x] 2b40df9955f9b5f04c74538565a426a28e20f873
- [x] d5e9bb14801e062a3f48c3525f1986383b0577c8
- [x] a3ae49a6c879f20eaa88d0c09b30a20ba7f91c81
- [x] f9bb48e56bf43479db80fa32349fb0c263dc673e
- [x] 3bfb0691bef350f2a2f03b097a0683773c347038

Please check the checkbox once you've added the changes from a commit to the Tripal 3 branch. **We have to go commit-by-commit due to the moving of many functions into classes and then renaming them.**

Once this is done I will use my function-by-function compare to make sure we didn't miss anything.

**Comment by laceysanderson** [comment]
Got started on this last night with commit 929d03b and 2a22113.
From now on, commit messages should read: `Issue #72: Updating API to include 2.1 improvements: commits AAA..BBB` where AAA and BBB are the first and second, respectively, 7.x-2.x commits you included in your changes.

For example, in my first commit I looked at 7.x-2.x commits cc79e48, 43077a9, a5ca25c, f6a5197, 4c8dc94, 9a34e75 and made sure all the changes from those commits were reflected in the 7.x-3.x branch. The commit message should have been `Issue #72: Updating API to include 2.1 improvements: commits cc79e48..9a34e75` which would make sure it shows up in this issue.

Also, you can just check the checkboxes above as you address a commit and the state of the checkboxes will be saved. This ensures we don't look at the same commits multiple times -we have enough on our plate without duplicating effort ;-p

**Comment by spficklin** [comment]
I'm working on these.  It's probably best to work bottom up so that we don't accidentally overwrite a fix with a previous fix in the same place that is older.

**Comment by spficklin** [comment]
I've run through all of the remaining.  It took me almost all day :-/  I cannot guarantee I didn't miss any or mess any up through human error..  But they've all been looked at.
